src/Compilers/Core/CodeAnalysisTest/Analyzers/AnalyzerConfigTests.cs

```diff
@@ -2485,6 +2485,57 @@ public void InvalidGlobalLevelIsIgnored()
             Assert.Equal(0, nonUserGlobalConfig.GlobalLevel);
         }
 
+        [Theory]
+        [InlineData("/dir1/dir3/../dir2/file.cs", true)]
+        [InlineData("/dir1/./././././dir2/file.cs", true)]
+        [InlineData("/dir1/../dir1/../dir1/../dir1/dir2/file.cs", true)]
+        [InlineData("/dir1/dir3/dir4/../dir2/file.cs", false)]
+        [InlineData("file.cs", false)]
+        [InlineData("", false)]
+        [InlineData("/../../dir1/dir2/file.cs", true)]
+        [InlineData("/./../dir1/dir2/file.cs", true)]
+        [InlineData("/dir1/../../dir1/dir2/file.cs", true)]
+        [InlineData("/..", false)]
+        [InlineData("/../file.cs", false)]
+        [InlineData("/dir1/../file.cs", false)]
+        [InlineData("./dir1/dir2/file.cs", false)]
+        [InlineData("././../.././dir1/dir2/file.cs", false)]
+        [InlineData("./dir1/../file.cs", false)]
+        [InlineData("../dir1/dir2.cs", false)]
+        public void EquivalentSourcePathNames(string sourcePath, bool shouldMatch)
```

Comment:
>public [](http://example.com/codeflow?start=8&length=6)

Consider testing `ExpandNormalizedFullPathWithRelativeParts()` directly and test UNC paths on Windows.

Reply:
Done. Explicitly excluding UNC paths. Its essentially a best effort, so if we can't expand it we'll just return the original string. Added tests for relative, device specified and UNC and called out in the remarks what we do.